### PR TITLE
add function setMapUrl to Router

### PR DESCRIPTION
**Pull request by RazzibShakya**

### CODE REVIEW

**Comment by RazzibShakya** [comment]
@syaau is it enough for a defaultRouter to have a URL Mapper?

**Comment by syaau** [comment]
Url Mapper function along with the initial url

File: src/router/Router.ts [hunk 1]
```diff
@@ -40,15 +40,19 @@ export class Router {
 
     this.stackSize = stackSize;
     this.mapUrl = mapUrl;
-
   }
+
+  setMapUrl = (mapUrl: UrlMapper) => {
+    if (this.mapUrl) return;
```
Comment (syaau) [inline src/router/Router.ts]: throw an error here, we don't want to silently ignore the setter. Also add initialUrl function to run the url mapper first time.

Comment (RazzibShakya) [reply src/router/Router.ts]: @syaau  do you mean the function will look like this??
```
  setMapUrl = (mapUrl: UrlMapper, initialUrl?: string) => {
    if (this.mapUrl) throw new Error('Url mapper has already been declared');
    this.mapUrl = mapUrl;
    if (initialUrl) {
      this.setUrl(initialUrl);
    }
  };

```

Comment (syaau) [reply src/router/Router.ts]: Yes, initialUrl needs to be set as this will be the main router initialization. The default initialUrl could be `/`.

**Comment by RazzibShakya** [comment]
@syaau  this is ready to merge.

**Review by syaau — APPROVED** [review]

**Pull request merged**